refactor(Refactor): Refactor project

gbzarelli opened the pull request:
# Changes

- Access level from beans (services/components)
- Make final in constructor arguments and methods
- Refactor some streams
- Private constructor in utilities classes
- Create an interface for UserService.java
- Change the find from UserRepository  to return an Optional

# Tests

- I don't change any tests to ensure its integrity. All tests passed

Took 1 hour 12 minutes

gbzarelli committed:
refactor(Refactor): Refactor project

- Access level from beans (services/components)
- Make final in
constructor arguments and methods
- Refactor some streams
- Private constructor in utilities classes
- Create an interface for UserService.java

Took 1 hour 12 minutes

Took 3 minutes

gbzarelli committed:
refactor(Refactor): Refactor project

- Access level from beans (services/components)
- Make final in
constructor arguments and methods
- Refactor some streams
- Private constructor in utilities classes
- Create an interface for UserService.java

Took 1 hour 12 minutes

lin-br commented on pom.xml:
```diff
@@ -10,7 +10,7 @@
     <groupId>br.com.tilmais</groupId>
     <artifactId>tilmhat</artifactId>
-    <version>0.0.0</version>
+    <version>0.0.1</version>
```
Acho que depois de toda essa mudança, seria melhor a versão 1.0.0 hein???

gbzarelli replied:
boa, vou inserir

gbzarelli replied:
feito @lin-br

lin-br commented on src/main/java/br/com/tilmais/tilmhat/controller/UserController.java:
```diff
@@ -12,26 +12,34 @@
-    public ResponseEntity register(@Valid @RequestBody UserRequestDTO dto) {
-        final Optional<URI> optionalURI = this.userService.registerUser(dto);
-        if (optionalURI.isPresent()) return ResponseEntity.created(optionalURI.get()).build();
+    public ResponseEntity<Serializable> register(@Valid @RequestBody final UserRequestDTO dto) {
```
Por que o retorno fica um `Serializable` ??

gbzarelli replied:
A definição da `ResponseEntity` requer um tipo, caso não passe ele irá ser `Object`, particularmente acho muito agressivo utilizar `Object` por isso acabo definindo um tipo mais conhecido, e que normamente os DTOs devem implementar que é o `Serializable`

 ```
public class ResponseEntity<T> extends HttpEntity<T> {
```

lin-br commented on src/main/java/br/com/tilmais/tilmhat/setting/ApplicationConstants.java:
```diff
@@ -1,6 +1,10 @@
 package br.com.tilmais.tilmhat.setting;
 
 public class ApplicationConstants {
```
Por que não uma `interface` ? Já que a classe possui somente as constantes, sem implementações, define ela como uma interface.

gbzarelli replied:
De acordo com o ilustrissimo livro Java Efetivo de J.Bloch seria um anti-pattern utilizar uma interface somente para definir constantes. Definições de constantes (significando estáticos, campos finais) devem ser colocadas em uma classe apropriada onde elas pertencem logicamente.
J. Bloch, Effective Java (third edition), Item 22. Addison-Wesley, 2008.

Pelo que eu entendi também o mais adequado é inserir essas constantes aonde fazem sentido; Vou analizar e fazer a alteração para você ver

lin-br replied:
Demorou, sobre o `Serializable` eu já entendi, obrigado liiin!

gbzarelli replied:
@lin-br feito, verifique como ficou, inserindo as constantes em suas classes responsáveis;

lin-br replied:
> @lin-br feito, verifique como ficou, inserindo as constantes em suas classes responsáveis;

Agora eu entendii kkkkkk
Top liin!

gbzarelli committed:
refactor(Refactor): Refactor project

- Removed the constant class and inserted their values in their responsibility
- Change version to 1.0.0

Took 33 minutes
